CMakeLists.txt

```diff
@@ -60,6 +60,28 @@ endif()
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -DDROP_CGAL ${CPP_STANDARD_FLAGS} -Wall -Wextra -Wpedantic -Wno-unused-variable -Wno-unused-parameter")
 
 
+#=====================================================================
+# check if the linker supports as needed or not
+#   (needed for ROOT dict, specifically on Ubuntu)
+set(STORE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+include(CheckCCompilerFlag)
+set(CMAKE_REQUIRED_FLAGS "-Wl,--no-as-needed")
+check_c_compiler_flag("" TEST_NO_AS_NEEDED)
+set(CMAKE_REQUIRED_FLAGS "-Wl,--as-needed")
+check_c_compiler_flag("" TEST_AS_NEEDED)
+set(CMAKE_REQUIRED_FLAGS ${STORE_REQUIRED_FLAGS})
+
+function(target_link_libraries_force_dict target dict)
+  if(TEST_NO_AS_NEEDED AND TEST_AS_NEEDED)
+    target_link_libraries(${target}
```

Comment:
I am not a fan of hiding these basic function calls. 
Also it might be more flexible to use the function just to add the push/pop flags around the argument and return the decorated result, this way one could still decide on the order of linked libraries without having the dict first

Reply:
Ah yes, that makes sense. So basically something like a `force_link_dict(dict)` that decorates the dictionary, such that in the end we could do:
```
force_link_dict(edm4hepDict)
target_link_library(read_events edm4hep edm4hepDict podio::podioRootIO)
```
which would leave the `target_link_library` call untouched and would also allow to force link more than one dictionary per call?